Add to planning refactoring

thecalcc opened the pull request:
SDESK-6267

Continuation of [this](https://github.com/superdesk/superdesk-planning/pull/1821) auto-closed PR

tomaskikutis committed:
adjust API

tomaskikutis committed:
update gettext usage

thecalcc committed:
Fix event template (#1812)

thecalcc committed:
Fix show modal (#1814)

thecalcc committed:
UIF version update (#1818)

thecalcc committed:
Add to planning action refactoring

tomaskikutis reviewed (changes requested):
fix conflicts, address comments from autoclosed PR

thecalcc committed:
Merge branch 'develop' into add-to-planning-refactoring

thecalcc commented:
Update package-json to use cc develop

thecalcc committed:
Fix lint

thecalcc committed:
Fix conditions

thecalcc committed:
Merge branch 'develop' into add-to-planning-refactoring

thecalcc committed:
Use gettext, remove duplicate code

thecalcc committed:
Fix import

thecalcc committed:
Merge branch 'develop' into add-to-planning-refactoring

thecalcc committed:
Small change, fix sd api usage

thecalcc committed:
Update sdapi usage

thecalcc committed:
Update client-core, use latest sd api

thecalcc committed:
Cleanup

thecalcc committed:
Merge branch 'develop' into add-to-planning-refactoring

thecalcc committed:
Fix imports

thecalcc committed:
Fix more imports

thecalcc committed:
Merge branch 'develop' into add-to-planning-refactoring

thecalcc committed:
Fix import

thecalcc committed:
Fix lint

tomaskikutis commented on client/planning-extension/src/extension.ts:
```diff
@@ -121,6 +121,31 @@ const extension: IExtension = {
                 entities: {
                     article: {
                         getActions: (item) => [
+                            {
```
If conditions aren't satisfied - the action should not be displayed. The way you implemented it - it is still displayed, but clicking it doesn't do anything :smile:

thecalcc replied:
I've come to the same realisation, that's why in the fulfil assignment action I added an `else` clause to notify the user they can't execute the action. For the future I can adjust the condition easily.

tomaskikutis replied:
What I don't get is why don't you modify `getActions` function so that if conditions aren't satisfied - it would return an empty array and no actions would be displayed to a user as a result?

tomaskikutis reviewed (commented):
Did you find out what is this about?

```
{
    action: 'external-app',
    type: 'addto-planning',
},
```

thecalcc commented:
> Did you find out what is this about?
> 
> ```
> {
>     action: 'external-app',
>     type: 'addto-planning',
> },
> ```

I'll address it in another PR, it's a filter for the actions. Basically responsible for displaying groups of actions.

thecalcc committed:
Condition based action rendering

thecalcc committed:
Planning authoring actions

thecalcc commented:
Based on a search across all repos I can confirm the impact of refactoring this part of the logic won't affect any customers:
<img width="1699" alt="image" src="https://github.com/user-attachments/assets/4d8a453d-b846-4036-95a6-3a1afb607004">

thecalcc committed:
Fix lint

tomaskikutis commented:
If `external-apps` extension point was used only in one place and we removed that usage is it safe to drop the entire extension point? If so, let's do it.

thecalcc commented:
> If `external-apps` extension point was used only in one place and we removed that usage is it safe to drop the entire extension point? If so, let's do it.

Don't mistake `external-app` for `external-appS`. There's only a single place in the code related to `external-app` which I haven't removed, as I removed all the registered objects matching this whole `external-app` code. Although I can remove the "connection" itself as well

tomaskikutis commented:
yeah, I meant removing `external-app` extension point. I think we can remove the entire `sdItemActionByIntent` directive from core.

thecalcc commented:
> yeah, I meant removing `external-app` extension point. I think we can remove the entire `sdItemActionByIntent` directive from core.

yep, i just removed it in client-core
